odoo/addons/base/models/ir_model.py

```diff
@@ -987,6 +987,10 @@ def create(self, vals_list):
         for vals in vals_list:
             if 'model_id' in vals:
                 vals['model'] = IrModel.browse(vals['model_id']).model
+                Model = self.env[vals['model']]
+                table_kind = tools.table_kind(self.env.cr, Model._table)
+                if table_kind != tools.TableKind.Regular:
+                    raise UserError(_('The model %s doesn\'t support adding fields.', Model._name))
```

Comment:
Actually, you can add computed fields.

This only applies if the field is stored.
Can you move this logic where we check the 'relation' field?
Both  of these conditions should be in an api.constrains functions in my opinion (example, a relation may be deleted later).